corporate/views/upgrade.py

```diff
@@ -89,66 +55,23 @@ def upgrade(
     ),
     licenses: Optional[int] = REQ(json_validator=check_int, default=None),
 ) -> HttpResponse:
-    customer = get_customer_by_realm(user.realm)
+    billing_session = RealmBillingSession(user)
+    customer = billing_session.get_customer()
     if customer is not None:
         ensure_customer_does_not_have_active_plan(customer)
```

Comment:
It is possible a further refactor could move the `if customer is not None` check into the `do_upgrade` call and thus not need to fetch the `customer` object in this request? I guess I'm not sure I know why it was outside the try/except block, but my guess is there's no reason in particular.

In any case, I think we can drop `Customer` from the `UpgradeRequest` dataclass, because it's obtained from the `billing_session` anyway.

Reply:
Yeah, I see that as a possible reason, but I don't see a clear reason why we'd have wanted that.